pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -166,7 +166,11 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 
 		typeConfig.Status.ObservedGeneration = typeConfig.Generation
 		typeConfig.Status.PropagationController = corev1a1.ControllerStatusNotRunning
-		typeConfig.Status.StatusController = corev1a1.ControllerStatusNotRunning
+
+		if typeConfig.Status.StatusController == nil {
+			typeConfig.Status.StatusController = new(corev1a1.ControllerStatus)
```

Comment:
(No action required) Ugh, I think `StatusController` and `PropagationController` are terrible names for fields indicating whether a controller is running or not.

Reply:
There will probably be another opportunity to update this post API review, but if we want to change these now is also a good time.